capa/features/extractors/elf.py

```diff
@@ -957,6 +960,485 @@ def guess_os_from_symtab(elf: ELF) -> Optional[OS]:
     return None
 
 
+def is_go_binary(elf: ELF) -> bool:
+    for shdr in elf.section_headers:
+        if shdr.get_name(elf) == ".note.go.buildid":
```

Comment:
The `go version` command uses logic similar to the one in the __get_go_buildinfo_date()__ function (in addition to looking for the BUILDINFO_MAGIC) to check if an executable is go or not. See references to the _errNotGoExe_ error here: https://github.com/golang/go/blob/master/src/debug/buildinfo/buildinfo.go#L41

Maybe consider incorporating that here? perhaps:
```suggestion
        if shdr.get_name(elf) in (".note.go.buildid", ".go.buildinfo"):
```

Or maybe consider retrieving the buildinfo and testing whether the BUILDINFO_MAGIC is therein? perhaps after we check the section names?

Also, I think this might be a good place to make use of [Google's Magika](https://google.github.io/magika/). Maybe if our heuristics can't determine the OS then output the error: **"Input file does not appear to target a supported OS. If you want to use Magika then run capa with the --use-magika parameter"**?

Reply:
`.go.buildinfo` checking added in https://github.com/mandiant/capa/pull/1987/commits/d54f8b50249e472606c15f97ec9ca060e3a6518d

Reply:
good idea on Magika, i will research that a bit.